### ISSUE TITLE
Create a component for out of the box LOD generation

### ISSUE THREAD

**Issue by Whinarn** [Whinarn/UnityMeshSimplifier#13]
Create a component for out of the box LOD (level of detail) generation.

The component must be easy to use and configure and allow to save the configuration for easy regeneration of the LODs when necessary.

**Comment by amirebrahimi** [comment]
I figured I'd share some reference if you decide to implement this internally. Here is what I do for AutoLOD:

https://github.com/Unity-Technologies/AutoLOD/blob/master/Scripts/Editor/LODData.cs (serializable asset for specific lod overrides)
https://github.com/Unity-Technologies/AutoLOD/blob/master/Scripts/Editor/LODDataEditor.cs (custom inspector)
https://github.com/Unity-Technologies/AutoLOD/blob/master/Scripts/Editor/LODImportSettings.cs (override-able import settings)
https://github.com/Unity-Technologies/AutoLOD/blob/master/Scripts/Editor/ModelImporterLODGenerator.cs (what controls the actual import)

**Comment by amirebrahimi** [comment]
Glad to see you're turning this into a package. I figure I can create an actual dependency and switch AutoLOD over to a package once yours lands, which is a better user experience then git submodules.

**Comment by Whinarn** [comment]
Thanks!

I have actually started work on this in the [lod-generator](https://github.com/Whinarn/UnityMeshSimplifier/tree/feature/lod-generator) branch based on my Unity examples from [MeshDecimator](https://github.com/Whinarn/MeshDecimator) but with some adjustments and rethinking. But I appreciate it! I will have a look to see if I can draw inspiration from AutoLOD.

My goal with this is to provide something easy to use without the need of code. I won't go much further than that.

Yes, I thought that it was a good idea to turn this into a Unity package instead of having it manually copied, or used as a git sub-module as you say. I can ping you once the release is ready. It might take a week or two since I'm doing this as a spare-time project 😄

**Comment by Godatplay** [comment]
Just wanted to say this is working well so far. Thank you :)

**Comment by Whinarn** [comment]
> Just wanted to say this is working well so far. Thank you :)

If you are using the work in progress branch, I recommend you to be careful since I cannot guarantee that it doesn't mess up your scenes or prefabs at the moment. I have a few more things that I want to take care of before I am finished with it. And then it requires a lot more testing.

I will try to have a release ready within 2 weeks from now. But I cannot make any promises on that.

**Comment by Whinarn** [comment]
As this has now been merged into the [develop](https://github.com/Whinarn/UnityMeshSimplifier/tree/develop) branch at commit https://github.com/Whinarn/UnityMeshSimplifier/commit/6528a0b96be48ab2b66d8d5c4b01a4a3a65571f0, I will close this issue.

A release containing this is currently being prepared.

**Comment by Whinarn** [comment]
@amirebrahimi I just release [v2.0.0](https://github.com/Whinarn/UnityMeshSimplifier/blob/master/CHANGELOG.md) of UnityMeshSimplifier that includes the structure and files needed for custom Unity packages.

Let me know if there's anything not working properly. I have performed tests locally here and it all seems to work fine, except that previous versions of Unity (2018) don't allow Git URLs in the manifest, so it would still have to be manually installed on those older versions.

**Comment by amirebrahimi** [comment]
Thanks @Whinarn - I just updated the submodule commit in AutoLOD; AutoLOD pretty much requires 2019.1 to get rid of warning spam from jobs, but I haven't set the min version to that yet; I'm also undecided on whether to turn it into a package yet, but if I do, then I can opt for using your package side-by-side instead of as a submodule.
